src/components/TheNav.vue

```diff
@@ -44,24 +41,22 @@
                         class="absolute top-0 left-0 right-0 h-screen bg-dark-yellow text-2xl"
                     >
                         <ul class="nav-items">
-                            <li
-                                @click="toggleNav(item.callback)"
-                                v-for="item in navItems"
-                                :key="item"
-                                v-show="isSignedIn === item.conditionOne
-                                || isSignedIn === item.conditionTwo"
-                            >
-                                <router-link
-                                    :to="item.path"
-                                    :class="{ active: $route.name === item.activeCondition }"
-                                    class="px-6 py-2">
-                                    {{ item.content }}
-                                </router-link>
-                            </li>
+                          <li
+                            @click="toggleNav(callback)"
+                            v-for="{ path, content, callback } in navItems"
+                            :key="path"
+                          >
+                            <router-link
+                                :to="path"
+                                :active-class="content !== 'Sign out' ? 'active' : ''"
+                                class="px-6 py-2">
+                                {{ content }}
+                            </router-link>
+                          </li>
                         </ul>
 
                         <img
-                            @click="toggleNav"
+                            @click="toggleNav()"
```

Comment:
This actually causes an error

Reply:
I believe it's actually quite the opposite - it was en error when clicking on this icon, before I added that. The error said `cb is not a function`. After I added this to the both images (here is the change only for one of them because I realized the second img lacked this as well after pushing changes, so I added it afterwards, locally) all errors dissapeared.

Reply:
It may have happened before, but that's the actual cause and I want you to know why. No matter if it disappeared or it didn't 😄

Reply:
Not sure, actually. I mean, I was convinced of the proper way of calling functions in Vue being `@click="callback"`, not `@click="callback()"` because this first option I was taught in my Vue course. However, when I wanted to pass two functions in some places of my code, I noticed the only way it worked was `@click="callback1(); callback2()"`. So I assumed I was supposed to have those parentheses when calling two functions and not have them when calling only one.

That's why I originally made `@click="toggleNav"`. But after adding this `@click="toggleNav(callback)"` to my nav items I noticed the mentioned error occuring but only when clicking those images with `@click="toggleNav"`. So I assumed that since this function accepts argument and the error mentioned cb, I should open those parentheses.

And this was actually my guess, so I don't know why this didn't worked and now it's working.

For me, it seems to be some Vue thing because in pure JS I would write:
`item.addEventListener('click', toggleNav)` not `item.addEventListener('click', toggleNav())`.

So I guess I'm lost 🙈

Reply:
If you have multiple funcitons to call it's better to wrap it with another (so called a handler).

```js
function handleClick() {
  doA();
  doB();
}

Reply:
in our case we can encapsulate toggleNav with an arrow function **in the template.**

```js
<some-element @click="() => doSth()"> 
```

Reply:
Notice that in both examples we have full control of the Event parameter which is passed to a function binded to a listener. 

`handleClick` can take the event and pass it down if needed.

Arrow function is just another way of doing kind of the same thing. It takes the event as the parameter, but returns another function call (in which we have full control again). So it could be sth like this:
```jsx
<div>{{ somethingElse }}</div>
<some-element @click="(event) => doSth(somethingElse)" />
```

Reply:
Corrected according to your comments but I don't think I understand the issue here. Apparently, you overestimated my cleverness 😅

Reply:
by passing it down I've meant:
```js
function handleClick(event) {
  doA(); // A doesn't need the event arg
  doB(event); // but B does
}
```

Reply:
If we want just to omit the event and put in it's place sth else, then we wrap our function with another function like this:
```js
// define our callback. - doB
function doB(awesomeParam) {
  return awesomeParam + 1
}

// define a wrapper for doB
function doA(someNotNeededEvent) {
  return doB(11) 
}

// doA will recieve an Event, but doB will be called with `11` set as awesomeParam
document.body.addEventListner('click', doA);
```

Reply:
Now you can see that parameter called `someNotNeededEvent` is not used in this code, so we can omit defining it:
```js
// define a wrapper for doB
function doA() {
  return doB(11) 
}
```
Works the same as before

Reply:
We can shorten this code even more by using the arrow fn notation inside the listener definition:
```js
document.body.addEventListner('click', () => doB(11));
```